src/ui/networkpage.rs

```diff
@@ -287,6 +292,35 @@ impl NetworkPage {
 
         StatefulWidget::render(list, list_rect, frame.buffer_mut(), &mut self.list.state);
     }
+
+    fn render_dpc_info(&mut self, model: &Rc<Model>, rect: Rect, frame: &mut Frame) {
+        let dpc_key = model.borrow().dpc_key.clone().unwrap_or("N/A".to_string());
+
+        let configuration_string = match dpc_key.as_str() {
+            "zedagent" => "From controller".green(),
+            "manual" => "Set by local user".yellow(),
```

Comment:
out of curiosity, .green(), yellow(), .red() are all mixin function you wrote for a &str?

Reply:
@uncleDecart no, this trait is imported from ratatui crate :) but yes, it is implemented as you think